### PR TITLE
Feature/api detail design

### PR DESCRIPTION
**Pull request by itsnameissteven**
## What does this PR do?
* More design has been added to the detailed API page.
* A back button added to the card
* A favorite button added to the card
* Reusable components have been broken out into respective files
* Favorite button was added to landing page api cards
* Users can save notes now and delete notes
* General cleanup around types and how they are passed 
-

## How should it be tested in the UI?

Fetch this branch review all refactored code code make sure everything works. 
Add notes to an api, the notes should persist. 
Delete notes, they should disappear, refresh page, they should remain deleted.
Add favorites. The favorite button should turn red to indicate it has been favorited. Anywhere in the app that the favorite button is present should have a red colored heart to indicate it has already been favorited. 
To remove from favorites click the button again. 
A list of favorites can be found in state when you use dev tools.

### Note
Code in the side bar component was commented out and will need to be adjusted once merged in. I wanted to avoid conflicts in work.

-

### CODE REVIEW

File: src/components/ApiCard/ApiCard.css [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
   border: 2px solid whitesmoke;
   text-align: left;
-  color: rgba(29, 29, 29, 0.986);
-  font-size: .5em;
+  font-size: 0.5em;
   width: 100%;
 }
 
+.api-card:hover {
+  box-shadow: -1px 3px 9px 1px rgba(0, 0, 0, 0.226);
```
Comment (Jacksonmcguire) [inline src/components/ApiCard/ApiCard.css:12]: Nice, the hover animations all around are looking sweet

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -111,27 +116,51 @@ class App extends React.Component<Props> {
...
+                apiList={this.state.currentApis} 
+                toggleFavorite={this.toggleFavorite} 
+                favorites={this.state.favorites}
+              />
+            </main>
+          )
         }}/>
         <Route
           path="/:title"
           render={({ match }) => {
```
Comment (Jacksonmcguire) [inline src/components/App/App.tsx:147]: might be worth breaking this out into it's own function to clean up the return a little bit

**Review by Jacksonmcguire — APPROVED** [review]
Left a few little comments but it's all looking good to go on my end, the favorite button and featured card / notes are looking great!

File: src/components/CardContainer/CardContainer.tsx [hunk 3]
```diff
@@ -4,21 +4,21 @@ import {ApiCard} from '../ApiCard/ApiCard'
...
-    key={index}
-    />
+    return (
+      <ApiCard
+        apiData={apiObj}
+        key={index}
+        toggleFavorite={toggleFavorite}
+        favorites={favorites}
+      />
+    )
```
Comment (cbmackintosh) [inline src/components/CardContainer/CardContainer.tsx:21]: Noticed a slight bug with the favoriting toggle. When there is only one favorited card in localStorage, clicking on the favorite heart does not remove it. In other words I can unfavorite all but one of my favorited cards.

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -111,27 +116,51 @@ class App extends React.Component<Props> {
...
+              <h1>metAPI</h1>
+              <FilterForm filter={this.filter} apiList={this.state.apiList}/>
+              <CardContainer 
+                apiList={this.state.currentApis} 
+                toggleFavorite={this.toggleFavorite} 
+                favorites={this.state.favorites}
+              />
+            </main>
+          )
         }}/>
```
Comment (cbmackintosh) [inline src/components/App/App.tsx:144]: Thanks for re-organizing this render method. Much more readable

File: src/components/FeaturedCard/FeaturedCard.tsx [hunk 5]
```diff
@@ -33,34 +43,39 @@ export class FeaturedCard extends Component<Props> {
...
-        <p>{this.props.HTTPS}</p>
-        <p>Category: {this.props.Category}</p>
-        <p>{this.props.Description}</p>
-        <p>
-          <a href={this.props.Link} target="_blank">
-            more details
+        <HomeLink to='/' className='home-link'>
+          <BiArrowBack className="home-link__arrow" />View all apis
+        </HomeLink>
+        <div className="favorite-btn-container">
```
Comment (cbmackintosh) [inline src/components/FeaturedCard/FeaturedCard.tsx:53]: Good call adding this return button on the Feature Card - improves navigability a lot. One thing we'll need to fix: If you conduct any kind of search, click on an API card and then click the 'view all' button, the search form on the homescreen is reset but only the search results are still visible.

**Review by cbmackintosh — APPROVED** [review]

**Comment by cbmackintosh** [comment]
Thanks for all the hard work on this branch. A few minor issues to fix here and there, but overall everything works perfectly.

**Pull request merged**